internal/templates/template.go

```diff
@@ -17,23 +19,20 @@ import (
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 )
 
-func Init(w io.Writer, template, dir string) error {
+func Init(w io.Writer, template, repo, subdir, dir string) error {
```

Comment:
Consider breaking this into `InitFromName` and `InitFromRepo` where `InitFromName` calls `InitFromRepo`. Also consider renaming variables to make them less ambiguous (dir, subdir, subdirPath, is pretty confusing. For example existing `dir` can be `target` or `targetDir`)

This would help simplify the logic and prevent all the duplicate variables you have here (e.g. subdir and subdirPath, repo and repoURL).

e.g.

```
# InitFromRepo is similar to existing `Init` but removes template map logic.
# it calls `strings.TrimSuifix` and can validate repo parses correctly.
func InitFromRepo(repo, subdir, target string) {
  ..
}

func InitFromName(w io.Writer, template, target string) {
  templatePath, ok := templates[template]
  if !ok {
    return usererr.New("unknown template %q", template)
  }
  return InitFromRepo("https://github.com/jetpack-io/devbox", templatePath, target)
}
```